modules/videoio/src/cap_gstreamer.cpp

```diff
@@ -216,8 +216,8 @@ class gst_initializer
         call_deinit = utils::getConfigurationParameterBool("OPENCV_VIDEOIO_GSTREAMER_CALL_DEINIT", false);
 
         GSafePtr<GError> err;
-        gst_init_check(NULL, NULL, err.getRef());
-        if (err)
+        gboolean gst_init_res = gst_init_check(NULL, NULL, err.getRef());
+        if (!gst_init_res)
         {
             CV_WARN("Can't initialize GStreamer: " << err->message);
```

Comment:
> err->message

Need to add NULL check here to avoid program crash: `err ? err->message : "<unknown reason>"`

Reply:
The documentation says:

> Parameters
> ...
> error – pointer to a GError to which a message will be posted on error

I was under the impression that if `gst_init_check` returns FALSE, then `err` has been set... is that wrong?

Reply:
Check CV_WARN line - it dereferences `err` object. This is invalid without NULL check. At least static code analyzers blame on this (they don't read the documentation).

Reply:
Got it! Thanks for the clarification!